src/constructs/aws/SinglePageApp.ts

```diff
@@ -0,0 +1,57 @@
+import * as cloudfront from "@aws-cdk/aws-cloudfront";
+import { FunctionEventType } from "@aws-cdk/aws-cloudfront";
+import type { Construct as CdkConstruct } from "@aws-cdk/core";
+import type { AwsProvider } from "@lift/providers";
+import type { FromSchema } from "json-schema-to-ts";
+import { getCfnFunctionAssociations } from "../../utils/getDefaultCfnFunctionAssociations";
+import { COMMON_STATIC_WEBSITE_DEFINITION, StaticWebsiteAbstract } from "./abstracts/StaticWebsiteAbstract";
+
+const SINGLE_PAGE_APP_DEFINITION = { ...COMMON_STATIC_WEBSITE_DEFINITION } as const;
+
+type Configuration = FromSchema<typeof SINGLE_PAGE_APP_DEFINITION>;
+
+export class SinglePageApp extends StaticWebsiteAbstract {
+    public static type = "single-page-app";
+    public static schema = SINGLE_PAGE_APP_DEFINITION;
+
+    constructor(
+        scope: CdkConstruct,
+        protected readonly id: string,
+        protected readonly configuration: Configuration,
+        protected readonly provider: AwsProvider
+    ) {
+        super(scope, id, configuration, provider);
+
+        const cfnDistribution = this.distribution.node.defaultChild as cloudfront.CfnDistribution;
+        const viewerRequestFunction = this.createViewerRequestCloudfrontFunction();
+
+        const defaultBehaviorFunctionAssociations = getCfnFunctionAssociations(cfnDistribution);
+
+        cfnDistribution.addOverride("Properties.DistributionConfig.DefaultCacheBehavior.FunctionAssociations", [
+            ...defaultBehaviorFunctionAssociations,
+            { EventType: FunctionEventType.VIEWER_REQUEST, FunctionARN: viewerRequestFunction.functionArn },
+        ]);
+    }
+
+    public createViewerRequestCloudfrontFunction(): cloudfront.Function {
+        /**
+         * CloudFront function that redirects nested paths to /index.html and
+         * let static files pass.
+         */
+        const code = `var REDIRECT_REGEX = /^[^.]+$|\\.(?!(css|gif|ico|jpg|js|png|txt|svg|woff|woff2|ttf|map|json)$)([^.]+$)/;
```

Comment:
Just to be 100% clear, we'll need to update this to whitelist new extensions that we may forget? (e.g. `jpeg` is not in there, I'm sure there are plenty we might not think about right?)

Reply:
Yes that's it.
I have taken the list from https://docs.aws.amazon.com/amplify/latest/userguide/redirects.html#redirects-for-single-page-web-apps-spa, as recommended by @fredericbarthelet in his [comment](https://github.com/getlift/lift/issues/172#issuecomment-1055175111).
In my opinion, we can trust amplify for the relevance of the list, wdyt ? (I'll add the `jpeg` though nice catch)

Reply:
Thanks, could you add a link to that URL in the comment (in the code)? That way we have a reference to check for updates. I think this makes a lot of sense to follow the AWS behavior 👍

Reply:
Done